pkg/apis/apps/v1beta1/defaults.go

```diff
@@ -30,6 +30,10 @@ func SetDefaults_StatefulSet(obj *StatefulSet) {
 	if len(obj.Spec.PodManagementPolicy) == 0 {
 		obj.Spec.PodManagementPolicy = OrderedReadyPodManagement
 	}
+
+	if obj.Spec.UpdateStrategy.Type == "" {
+		obj.Spec.UpdateStrategy.Type = RollingUpdateStatefulSetStrategyType
```

Comment:
The whole point of having "OnDelete" as a strategy is to be backwards-compatible. This is a breaking change.

cc: @kubernetes/sig-apps-api-reviews